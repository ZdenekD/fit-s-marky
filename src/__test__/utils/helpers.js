import React from 'react';
import {shallow} from 'enzyme';

/**
 * Factory function to create a shallow wrapper for App component
 * @function setup
 * @param {ReactComponent} Component - React component to render with shallow
 * @param {Object} props - Component props specific to this setup
 * @returns {ShallowWrapper}
 */
export const factory = (Component, props = {}) => {
    const wrapper = shallow(<Component {...props} />);

    return wrapper;
};

/**
 * Return ShallowWrapper containg node(s) with the data-test value
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within
 * @param {string} value - data attribute value to find
 * @returns {ShallowWrapper}
 */
export const findComponent = (wrapper, value) => wrapper.find(`[data-test="${value}"]`);
