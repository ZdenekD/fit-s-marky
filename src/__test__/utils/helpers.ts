import {ShallowWrapper} from 'enzyme';

const findComponent = (wrapper: ShallowWrapper, value: string): ShallowWrapper => wrapper.find(`[data-test="${value}"]`);

export default findComponent;
