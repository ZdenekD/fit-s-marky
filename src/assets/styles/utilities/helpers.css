:global ._pull-right {
    float: right !important;
}

:global ._pull-left {
    float: left !important;
}

:global ._italic {
    font-style: italic;
}
