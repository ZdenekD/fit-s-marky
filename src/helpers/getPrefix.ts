const getPrefix = (): string => '_5innv';

export default getPrefix;
