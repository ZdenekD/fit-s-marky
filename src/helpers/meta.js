import React from 'react';

const getMeta = item => (
    [
        <meta name="description" content={item.description} key={item.id} />,
        <meta property="og:title" content={item.title} key={item.id} />,
        <meta property="og:type" content="website" key={item.id} />,
        <meta property="og:description" content={item.description} key={item.id} />,
        <meta property="og:url" content="https://fit-s-marky.cz/" key={item.id} />,
        <meta name="twitter:card" content="summary" key={item.id} />,
        <meta name="twitter:title" content={item.title} key={item.id} />,
        <meta name="twitter:description" content={item.description} key={item.id} />,
        <meta name="twitter:domain" content="https://fit-s-marky.cz/" key={item.id} />,
    ]
);

export default getMeta;
