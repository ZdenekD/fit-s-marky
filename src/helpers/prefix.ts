const prefix = '_5innv';

export default prefix;
