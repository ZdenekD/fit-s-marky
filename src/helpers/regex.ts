interface IRegex {
    text: RegExp
    zip: RegExp
    email: RegExp
}

const regex: IRegex = {
    text: /^[éěëščřžýáäíïťďňöúůüÿÉĚËŠČŘŽÝÁÄÍÏŤĎŇÖÚŮÜŸ \-,.?!'"\w\s\d]+$/,
    zip: /^\d{3}(\s)?\d{2}$/,
    email: /^("([ !#-[\]-~]|\\[ -~])+"|[-a-z0-9!#$%&'*+/=?^_`{|}~]+(\.[-a-z0-9!#$%&'*+/=?^_`{|}~]+)*)@([0-9a-z\u00C0-\u02FF\u0370-\u1EFF]([-0-9a-z\u00C0-\u02FF\u0370-\u1EFF]{0,61}[0-9a-z\u00C0-\u02FF\u0370-\u1EFF])?\.)+[a-z\u00C0-\u02FF\u0370-\u1EFF]([-0-9a-z\u00C0-\u02FF\u0370-\u1EFF]{0,17}[a-z\u00C0-\u02FF\u0370-\u1EFF])?$/i,
};

export default regex;
