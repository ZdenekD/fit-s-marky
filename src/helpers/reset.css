a,
abbr,
address,
article,
aside,
audio,
b,
blockquote,
body,
button,
canvas,
caption,
cite,
code,
dd,
del,
details,
dfn,
div,
dl,
dt,
em,
embed,
fieldset,
figcaption,
figure,
footer,
form,
h1,
h2,
h3,
h4,
h5,
h6,
header,
html,
i,
iframe,
img,
input,
ins,
kbd,
label,
legend,
li,
main,
mark,
nav,
object,
ol,
output,
p,
pre,
q,
ruby,
s,
samp,
section,
select,
small,
span,
strong,
sub,
summary,
sup,
table,
tbody,
td,
textarea,
tfoot,
th,
thead,
time,
tr,
u,
ul,
var,
video {
    margin: 0;
    padding: 0;
    vertical-align: baseline;
    background: transparent;
    font: inherit;
    font-size: 100%;
    outline: 0;
    border: none;
}

*,
*::after,
*::before {
    box-sizing: border-box;
}

body {
    height: 100%;
    overflow-y: scroll;
    line-height: 1;
}

article,
aside,
details,
figcaption,
figure,
footer,
header,
main,
nav,
section {
    display: block;
}

audio,
canvas,
video {
    display: inline-block;
}

ol,
ul {
    list-style: none;
}
