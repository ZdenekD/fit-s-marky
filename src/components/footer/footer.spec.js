import {factory, findComponent} from '../../__test__/utils/helpers';
import Footer from './index';

describe('Footer', () => {
    it('renders without error', () => {
        const wrapper = factory(Footer);
        const component = findComponent(wrapper, 'component-footer');

        expect(component.length).toBe(1);
    });
});
