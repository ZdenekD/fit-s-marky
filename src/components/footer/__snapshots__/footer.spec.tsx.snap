// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer match snapshot 1`] = `
<footer class="footer"
        data-test="component-footer"
>
  <section class="section"
           data-test="component-clients"
           style="opacity:0"
  >
    <h2 class="heading">
      Spokojení klienti:
    </h2>
    <ul class="list">
      <li class="item">
        <img src="/clients/johnson&amp;johnson.png"
             alt
             class="image"
        >
        <br>
        <span class="title">
          Johnson&amp;Johnson
        </span>
      </li>
      <li class="item">
        <img src="/clients/caerus_im.png"
             alt
             class="image"
        >
        <br>
        <span class="title">
          Caerus IM
        </span>
      </li>
      <li class="item">
        <img src="/clients/vodafone.png"
             alt
             class="image"
        >
        <br>
        <span class="title">
          Vodafone
        </span>
      </li>
      <li class="item">
        <img src="/clients/paralelni_polis.png"
             alt
             class="image"
        >
        <br>
        <span class="title">
          Paralelní Polis
        </span>
      </li>
      <li class="item">
        <img src="/clients/form_factory.png"
             alt
             class="image"
        >
        <br>
        <span class="title">
          Form Factory
        </span>
      </li>
      <li class="item">
        <img src="/clients/pilates&amp;flow.png"
             alt
             class="image"
        >
        <br>
        <span class="title">
          Pilates &amp; Flow
        </span>
      </li>
      <li class="item">
        <img src="/clients/joga_letna.png"
             alt
             class="image"
        >
        <br>
        <span class="title">
          Joga Letná
        </span>
      </li>
      <li class="item">
        <img src="/clients/fitness_flora.png"
             alt
             class="image"
        >
        <br>
        <span class="title">
          HitFit Flora
        </span>
      </li>
      <li class="item">
        <img src="/clients/k6.png"
             alt
             class="image"
        >
        <br>
        <span class="title">
          Pohybové studio K6
        </span>
      </li>
      <li class="item">
        <img src="/clients/obvodni_soud.png"
             alt
             class="image"
        >
        <br>
        <span class="title">
          Obvodní soud pro Prahu 1
        </span>
      </li>
      <li class="item">
        <img src="/clients/sokol_jahodnice.png"
             alt
             class="image"
        >
        <br>
        <span class="title">
          TJ Sokol Jahodnice
        </span>
      </li>
    </ul>
  </section>
  <section class="rights">
    <svg class="icon icon-tabler icon-tabler-copyright"
         width="16"
         height="16"
         viewbox="0 0 24 24"
         stroke-width="1"
         stroke="currentColor"
         fill="none"
         stroke-linecap="round"
         stroke-linejoin="round"
    >
      <path stroke="none"
            d="M0 0h24v24H0z"
            fill="none"
      >
      </path>
      <circle cx="12"
              cy="12"
              r="9"
      >
      </circle>
      <path d="M14.5 9a3.5 4 0 1 0 0 6">
      </path>
    </svg>
    Fit s Marky | All Rights Reserved
  </section>
  <section class="contact">
    <span class="link">
      <svg class="icon icon-tabler icon-tabler-phone"
           width="20"
           height="20"
           viewbox="0 0 24 24"
           stroke-width="1"
           stroke="currentColor"
           fill="none"
           stroke-linecap="round"
           stroke-linejoin="round"
      >
        <path stroke="none"
              d="M0 0h24v24H0z"
              fill="none"
        >
        </path>
        <path d="M5 4h4l2 5l-2.5 1.5a11 11 0 0 0 5 5l1.5 -2.5l5 2v4a2 2 0 0 1 -2 2a16 16 0 0 1 -15 -15a2 2 0 0 1 2 -2">
        </path>
      </svg>
      <a href="tel:728618337"
         title="Telefon"
         class="anchor "
         data-test="component-anchor"
      >
        728 618 337
      </a>
    </span>
    <span class="link">
      <svg class="icon icon-tabler icon-tabler-mail"
           width="20"
           height="20"
           viewbox="0 0 24 24"
           stroke-width="1"
           stroke="currentColor"
           fill="none"
           stroke-linecap="round"
           stroke-linejoin="round"
      >
        <path stroke="none"
              d="M0 0h24v24H0z"
              fill="none"
        >
        </path>
        <rect x="3"
              y="5"
              width="18"
              height="14"
              rx="2"
        >
        </rect>
        <polyline points="3 7 12 13 21 7">
        </polyline>
      </svg>
      <a href="mailto:fitsmarky@seznam.cz"
         title="E-mail: fitsmarky@seznam.cz"
         class="anchor "
         data-test="component-anchor"
      >
        fitsmarky@seznam.cz
      </a>
    </span>
    <span class="link">
      <a href="https://www.facebook.com/fitsmarky"
         title="Facebook"
         class="anchor social"
         target="_blank"
         rel="noreferrer noopener"
         data-test="component-anchor"
      >
        <svg class="icon icon-tabler icon-tabler-thumb-up"
             width="28"
             height="28"
             viewbox="0 0 24 24"
             stroke-width="1"
             stroke="currentColor"
             fill="none"
             stroke-linecap="round"
             stroke-linejoin="round"
        >
          <path stroke="none"
                d="M0 0h24v24H0z"
                fill="none"
          >
          </path>
          <path d="M7 11v8a1 1 0 0 1 -1 1h-2a1 1 0 0 1 -1 -1v-7a1 1 0 0 1 1 -1h3a4 4 0 0 0 4 -4v-1a2 2 0 0 1 4 0v5h3a2 2 0 0 1 2 2l-1 5a2 3 0 0 1 -2 2h-7a3 3 0 0 1 -3 -3">
          </path>
        </svg>
      </a>
    </span>
  </section>
</footer>
`;
